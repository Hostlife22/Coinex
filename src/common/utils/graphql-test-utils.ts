interface IReq {
  alias?: string;
  body?: any;
}

export const hasOperationName = (req: IReq, operationName: string) => {
  const { body } = req;
  return body.hasOwnProperty('operationName') && body.operationName === operationName;
};

export const aliasQuery = (req: IReq, operationName: string) => {
  if (hasOperationName(req, operationName)) {
    req.alias = `gql${operationName}Query`;
  }
};

export const aliasMutation = (req: IReq, operationName: string) => {
  if (hasOperationName(req, operationName)) {
    req.alias = `gql${operationName}Mutation`;
  }
};
