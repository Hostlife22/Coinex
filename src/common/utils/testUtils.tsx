import { ReactNode } from 'react';
import { Provider } from 'react-redux';
import { MemoryRouter, Route, Routes } from 'react-router-dom';
import { setupApiStore } from '../../app/api/setupApiStore';
import { userApi } from '../../app/api/userApi';
import authReducer from '../../features/auth/authSlice';

interface IProps {
  children: ReactNode;
}

interface IRouterProvider extends IProps {
  route: string;
}

export const Wrapper = ({ children }: IProps) => {
  const storeRef = setupApiStore(userApi, { auth: authReducer });
  return <Provider store={storeRef.store}>{children}</Provider>;
};

export const RouterProvider = ({ children, route }: IRouterProvider) => (
  <MemoryRouter initialEntries={[route]}>
    <Routes>
      <Route path={route} element={children} />
    </Routes>
  </MemoryRouter>
);

export const ElemProvider = ({ children, route }: IRouterProvider) => {
  return (
    <Wrapper>
      <RouterProvider route={route}>{children}</RouterProvider>
    </Wrapper>
  );
};
