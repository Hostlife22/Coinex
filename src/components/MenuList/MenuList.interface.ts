import { DetailedHTMLProps, HTMLAttributes } from 'react';

export interface IMenuListProps
  extends DetailedHTMLProps<HTMLAttributes<HTMLUListElement>, HTMLUListElement> {}
