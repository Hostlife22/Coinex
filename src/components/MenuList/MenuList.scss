.menu-list {
  &__title {
    min-width: 60px;
    color: var(--color-light-gray);
    border-radius: 6px;
    transition: all 0.3s ease;
  }

  &__link {
    text-decoration: none;

    li {
      display: flex;
      align-items: center;
      height: 40px;
      margin-top: 10px;
      list-style: none;
      border-radius: 6px;

      &:hover {
        background-color: var(--color-primary);

        .menu-list__title {
          color: var(--bg-black-700);
        }

        > span {
          color: var(--bg-black-700);
        }
      }

      > span {
        display: flex;
        align-items: center;
        justify-content: center;
        min-width: 60px;
        height: 100%;
        color: var(--color-light-gray);
        font-size: 20px;
        border-radius: 6px;
        transition: all 0.3s ease;
      }
    }

    &_active {
      li {
        background-color: var(--color-primary);

        .menu-list__title {
          color: var(--bg-black-700);
        }

        span {
          color: var(--bg-black-700);
        }
      }
    }
  }
}

@media screen and (max-width: 1024px) {
  .menu-list {
    &__link li {
      > span {
        min-width: 45px;
      }
    }
  }
}
