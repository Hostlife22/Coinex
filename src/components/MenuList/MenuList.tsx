import cn from 'classnames';
import { NavLink } from 'react-router-dom';
import { menuList } from '../../common/data';
import Ptag from '../Ptag/Ptag';
import { IMenuListProps } from './MenuList.interface';
import './MenuList.scss';

const MenuList = ({ className, ...props }: IMenuListProps) => {
  return (
    <ul className={cn(className, 'menu-list')} {...props}>
      {menuList.map((item) => (
        <NavLink
          key={item.id}
          to={item.route}
          className={({ isActive }) =>
            isActive ? 'menu-list__link menu-list__link_active' : 'menu-list__link'
          }>
          <li>
            <span>{item.icon}</span>
            <Ptag className="menu-list__title" size="l">
              {item.title}
            </Ptag>
          </li>
        </NavLink>
      ))}
    </ul>
  );
};

export default MenuList;
