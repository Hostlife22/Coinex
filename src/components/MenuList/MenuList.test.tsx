import { render, screen } from '@testing-library/react';
import { ElemProvider } from '../../common/utils/testUtils';
import MenuList from './MenuList';

describe('MenuList component', () => {
  it('renders correctly', () => {
    render(
      <ElemProvider route="/">
        <MenuList />
      </ElemProvider>,
    );

    expect(screen.getByText(/dashboard/i)).toBeInTheDocument();
    expect(screen.getByText(/revenue/i)).toBeInTheDocument();
    expect(screen.getByText(/wallet/i)).toBeInTheDocument();
  });

  it('renders correctly with custom className', () => {
    render(
      <ElemProvider route="/">
        <MenuList />
      </ElemProvider>,
    );

    const activeLink = screen.getByRole('link', { current: 'page' });

    expect(activeLink).toBeInTheDocument();
    expect(activeLink).toHaveTextContent(/dashboard/i);
  });

  it('MenuList snapshot', () => {
    const view = render(
      <ElemProvider route="/">
        <MenuList />
      </ElemProvider>,
    );

    expect(view).toMatchSnapshot();
  });
});
