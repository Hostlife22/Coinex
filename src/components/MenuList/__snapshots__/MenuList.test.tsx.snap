// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MenuList component MenuList snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <ul
        class="menu-list"
      >
        <a
          aria-current="page"
          class="menu-list__link menu-list__link_active"
          href="/"
        >
          <li>
            <span>
              <svg
                fill="currentColor"
                height="1em"
                stroke="currentColor"
                stroke-width="0"
                viewBox="0 0 576 512"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M280.37 148.26L96 300.11V464a16 16 0 0 0 16 16l112.06-.29a16 16 0 0 0 15.92-16V368a16 16 0 0 1 16-16h64a16 16 0 0 1 16 16v95.64a16 16 0 0 0 16 16.05L464 480a16 16 0 0 0 16-16V300L295.67 148.26a12.19 12.19 0 0 0-15.3 0zM571.6 251.47L488 182.56V44.05a12 12 0 0 0-12-12h-56a12 12 0 0 0-12 12v72.61L318.47 43a48 48 0 0 0-61 0L4.34 251.47a12 12 0 0 0-1.6 16.9l25.5 31A12 12 0 0 0 45.15 301l235.22-193.74a12.19 12.19 0 0 1 15.3 0L530.9 301a12 12 0 0 0 16.9-1.6l25.5-31a12 12 0 0 0-1.7-16.93z"
                />
              </svg>
            </span>
            <p
              class="menu-list__title large-text"
            >
              Dashboard
            </p>
          </li>
        </a>
        <a
          class="menu-list__link"
          href="/revenue"
        >
          <li>
            <span>
              <svg
                fill="currentColor"
                height="1em"
                stroke="currentColor"
                stroke-width="0"
                viewBox="0 0 512 512"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M500 384c6.6 0 12 5.4 12 12v40c0 6.6-5.4 12-12 12H12c-6.6 0-12-5.4-12-12V76c0-6.6 5.4-12 12-12h40c6.6 0 12 5.4 12 12v308h436zM372.7 159.5L288 216l-85.3-113.7c-5.1-6.8-15.5-6.3-19.9 1L96 248v104h384l-89.9-187.8c-3.2-6.5-11.4-8.7-17.4-4.7z"
                />
              </svg>
            </span>
            <p
              class="menu-list__title large-text"
            >
              Revenue
            </p>
          </li>
        </a>
        <a
          class="menu-list__link"
          href="/wallet"
        >
          <li>
            <span>
              <svg
                fill="currentColor"
                height="1em"
                stroke="currentColor"
                stroke-width="0"
                viewBox="0 0 512 512"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M461.2 128H80c-8.84 0-16-7.16-16-16s7.16-16 16-16h384c8.84 0 16-7.16 16-16 0-26.51-21.49-48-48-48H64C28.65 32 0 60.65 0 96v320c0 35.35 28.65 64 64 64h397.2c28.02 0 50.8-21.53 50.8-48V176c0-26.47-22.78-48-50.8-48zM416 336c-17.67 0-32-14.33-32-32s14.33-32 32-32 32 14.33 32 32-14.33 32-32 32z"
                />
              </svg>
            </span>
            <p
              class="menu-list__title large-text"
            >
              Wallet
            </p>
          </li>
        </a>
      </ul>
    </div>
  </body>,
  "container": <div>
    <ul
      class="menu-list"
    >
      <a
        aria-current="page"
        class="menu-list__link menu-list__link_active"
        href="/"
      >
        <li>
          <span>
            <svg
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 576 512"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M280.37 148.26L96 300.11V464a16 16 0 0 0 16 16l112.06-.29a16 16 0 0 0 15.92-16V368a16 16 0 0 1 16-16h64a16 16 0 0 1 16 16v95.64a16 16 0 0 0 16 16.05L464 480a16 16 0 0 0 16-16V300L295.67 148.26a12.19 12.19 0 0 0-15.3 0zM571.6 251.47L488 182.56V44.05a12 12 0 0 0-12-12h-56a12 12 0 0 0-12 12v72.61L318.47 43a48 48 0 0 0-61 0L4.34 251.47a12 12 0 0 0-1.6 16.9l25.5 31A12 12 0 0 0 45.15 301l235.22-193.74a12.19 12.19 0 0 1 15.3 0L530.9 301a12 12 0 0 0 16.9-1.6l25.5-31a12 12 0 0 0-1.7-16.93z"
              />
            </svg>
          </span>
          <p
            class="menu-list__title large-text"
          >
            Dashboard
          </p>
        </li>
      </a>
      <a
        class="menu-list__link"
        href="/revenue"
      >
        <li>
          <span>
            <svg
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 512 512"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M500 384c6.6 0 12 5.4 12 12v40c0 6.6-5.4 12-12 12H12c-6.6 0-12-5.4-12-12V76c0-6.6 5.4-12 12-12h40c6.6 0 12 5.4 12 12v308h436zM372.7 159.5L288 216l-85.3-113.7c-5.1-6.8-15.5-6.3-19.9 1L96 248v104h384l-89.9-187.8c-3.2-6.5-11.4-8.7-17.4-4.7z"
              />
            </svg>
          </span>
          <p
            class="menu-list__title large-text"
          >
            Revenue
          </p>
        </li>
      </a>
      <a
        class="menu-list__link"
        href="/wallet"
      >
        <li>
          <span>
            <svg
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 512 512"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M461.2 128H80c-8.84 0-16-7.16-16-16s7.16-16 16-16h384c8.84 0 16-7.16 16-16 0-26.51-21.49-48-48-48H64C28.65 32 0 60.65 0 96v320c0 35.35 28.65 64 64 64h397.2c28.02 0 50.8-21.53 50.8-48V176c0-26.47-22.78-48-50.8-48zM416 336c-17.67 0-32-14.33-32-32s14.33-32 32-32 32 14.33 32 32-14.33 32-32 32z"
              />
            </svg>
          </span>
          <p
            class="menu-list__title large-text"
          >
            Wallet
          </p>
        </li>
      </a>
    </ul>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
